wallet/walletrpc.c

```diff
@@ -218,26 +111,21 @@ static void json_withdraw(struct command *cmd,
 	}
 
 	/* Parse address. */
-	withdraw->destination
-		= scriptpubkey_from_address(withdraw, &testnet,
-					    buffer + desttok->start,
-					    desttok->end - desttok->start);
+	bip173 = json_tok_address_scriptpubkey(cmd, buffer,
+					       desttok, (const u8**)(&withdraw->destination));
 
 	/* Check that destination address could be understood. */
-	if (!withdraw->destination) {
+	if (!bip173) {
 		command_fail(cmd, "Could not parse destination address");
 		return;
 	}
 
 	/* Check address given is compatible with the chain we are on. */
-	if (testnet != get_chainparams(cmd->ld)->testnet) {
-		if (testnet) {
-			command_fail(cmd,
-				    "Use of testnet address on mainnet");
-		} else {
-			command_fail(cmd,
-				    "Use of mainnet address on testnet");
-		}
+	if (chainparams_by_bip173(bip173)->testnet != get_chainparams(cmd->ld)->testnet) {
```

Comment:
Should be `streq(bip173, get_chainparams(cmd->ld)->bip173_name)` instead.

Reply:
I actually did that first but noticed that `test_withdraw` fails.
This is because we generate testnet addresses but the network is regtest.

Reply:
Yes, we also need to update the test to use proper `bcrt1` addreses. `bcrt1qw508d6qejxtdg4y5r3zarvary0c5xw7kygt080`for P2WPKH, `bcrt1qrp33g0q5c5txsp9arysrx4k6zdkfs4nce4xj0gdcccefvpysxf3qzf4jry` for P2WSH.

Reply:
ohh, right, thanks. Will look into that.